agent/src/main/java/com/code_intelligence/jazzer/autofuzz/FuzzTarget.java

```diff
@@ -84,19 +84,28 @@ public static void fuzzerInitialize(String[] args) {
       descriptor = null;
     }
 
-    Class<?> targetClass;
-    try {
-      // Explicitly invoking static initializers to trigger some coverage in the code.
-      targetClass = Class.forName(className, true, ClassLoader.getSystemClassLoader());
-    } catch (ClassNotFoundException e) {
-      System.err.printf(
-          "Failed to find class %s for autofuzz, please ensure it is contained in the classpath "
-              + "specified with --cp and specify the full package name%n",
-          className);
-      e.printStackTrace();
-      System.exit(1);
-      return;
-    }
+    Class<?> targetClass = null;
+    String targetClassName = className;
+    do {
```

Comment:
What do you think of the alternative of replacing all `.` that precede a capital letter (except the first) with `$` and then only trying once? The current approach isn't bad, I'm just wondering whether it could cause any ambiguities.

Reply:
Class names _could_ be lowercase and package names _could_ be uppercase. Probably no one is going to do that, though. I think the current approach is a little bit more on the safe side.

Reply:
True, especially for other JVM-based languages.